src/beamlime/logging/loggers.py

```diff
@@ -0,0 +1,127 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+
+import logging
+import sys
+from typing import Any, Callable, Union
+
+
+def hold_logging(logging_func: Callable) -> Callable:
+    from functools import wraps
+    from inspect import signature
+
+    @wraps(logging_func)
+    def wrapper(*args: Any, **kwargs: Any) -> Any:
+        logging._acquireLock()
```

Comment:
Is there are way to do this without using protected attributes of `logging`?

Reply:
I updated not to use them...!